src/remodel_api/remodel.rs

```diff
@@ -28,54 +30,43 @@ fn xml_decode_options() -> rbx_xml::DecodeOptions {
     rbx_xml::DecodeOptions::new().property_behavior(rbx_xml::DecodePropertyBehavior::ReadUnknown)
 }
 
-fn get_required_string_option(options: &Table, option: &str) -> rlua::Result<String> {
+fn get_required_option<'lua, T: FromLua<'lua>>(
+    context: Context<'lua>,
+    options: &Table<'lua>,
+    option: &str,
+) -> rlua::Result<T> {
     let value = options.get(option).map_err(rlua::Error::external)?;
 
-    match value {
-        Value::String(value) => Ok(value.to_str().map_err(rlua::Error::external)?.to_string()),
-        Value::Nil => Err(rlua::Error::external(format!(
+    if let Value::Nil = value {
+        Err(rlua::Error::external(format!(
             "The option {} must be specified",
             option
-        ))),
-        _ => Err(rlua::Error::external(format!(
-            "The option {} must be a string",
-            option
-        ))),
-    }
-}
-
-fn get_string_option(options: &Table, option: &str, default: &str) -> rlua::Result<String> {
-    let value = options.get(option).map_err(rlua::Error::external)?;
-
-    match value {
-        Value::String(value) => Ok(value.to_str().map_err(rlua::Error::external)?.to_string()),
-        Value::Nil => Ok(default.to_string()),
-        _ => Err(rlua::Error::external(format!(
-            "The option {} must be a string",
-            option
-        ))),
+        )))
+    } else {
+        FromLua::from_lua(value, context)
     }
 }
 
-fn get_bool_option(options: &Table, option: &str, default: bool) -> rlua::Result<bool> {
+fn get_option<'lua, T: FromLua<'lua>>(
+    context: Context<'lua>,
+    options: &Table<'lua>,
+    option: &str,
+    default: T,
```

Comment:
Let's just make this function return an `rlua::Result<Option<T>>` instead of passing a default. Rust has lots of great features that let us easily fall back to a default that are more flexible than this. We can also merge it with `get_required_option` for the same reason!